include/pybind11/pybind11.h

```diff
@@ -1910,7 +1910,7 @@ template <return_value_policy Policy = return_value_policy::reference_internal,
 template <typename InputType, typename OutputType> void implicitly_convertible() {
     struct set_flag {
         bool &flag;
-        set_flag(bool &flag) : flag(flag) { flag = true; }
+        set_flag(bool &my_flag) : flag(my_flag) { my_flag = true; }
```

Comment:
`my_flag` sounds very tutorial-for-programming-like, though. Could you change this to `flag_` or `f` or so (or suggest something else)?

Also (mostly to other reviewers): should we turn on more warning flags in CI? What's our goal when it comes to such warnings warnings?

Reply:
I figured you would have a preferred naming for something like this, I just couldn't find it. I use `m for member` style (`mFlag`) so I never have to deal with this in my own code.

I'll use `flag_`.

Reply:
Thanks! I can't really find it either. We do use `m_flag` for private class members, but this is a struct, so `flag` is part of the interface (well, it's a local struct in a function, so it's not that important, but you don't want to write `some_struct.m_flag`, I think :-) )

Reply:
A quick look seems that `f` and `flag_` would be the two options. I found instances of both conventions. I think this is fine, though, so thanks!